src/composite_core.jl

```diff
@@ -0,0 +1,124 @@
+struct PrimalAdditionFailedException{P} <: Exception
+    primal::P
+    differential::Composite{P}
+    original::Exception
+end
+
+function Base.showerror(io::IO, err::PrimalAdditionFailedException{P}) where {P}
+    println(io, "Could not construct $P after addition")
```

Comment:
why is this to `io` but the others not?

Reply:
OH, that is why everything is coming out wrong and out order.